test/__fixtures__/parameter-decorator-typed/output.js

```diff
@@ -29,18 +29,23 @@ let MyOtherClass = (_dec4 = function (target, key) {
 
   method(param, schema) {}
 
-}, (_applyDecoratedDescriptor(_class3.prototype, "methodUndecorated", [_dec8, _dec9, _dec10], Object.getOwnPropertyDescriptor(_class3.prototype, "methodUndecorated"), _class3.prototype), _applyDecoratedDescriptor(_class3.prototype, "method", [_dec11, _dec12, _dec13, _dec14, _dec15], Object.getOwnPropertyDescriptor(_class3.prototype, "method"), _class3.prototype)), _class3)) || _class2) || _class2) || _class2) || _class2);
-let DecoratedClass = (_dec16 = function (target, key) {
+  methodWithObjectSpread(@argObjectSpread()
```

Comment:
The decorator should be gone after the transpiling step, seems very strange to me. Have you tried running the code? I think this would break anyway users code

Reply:
good pickup, yes that shouldn't be there.
Do you know which code removes the `Identifier` decorator after adding the metadata calls?

Reply:
Right now param decorators are erased in the `parameterVisitor` (here: https://github.com/leonardfactory/babel-plugin-transform-typescript-metadata/blob/775caf3244a610efa3d8bfb5408a366b7a1fe321/src/parameter/parameterVisitor.ts#L94), probably the destructuring assignment is changing the node where the decorator is applied. We should be able to inspect it with an AST viewer and identify the correct node to erase. Let me know!
PS: Thank you for the PR 🙏🏼

Reply:
Think I have got it now.

After reading the generated code it's clear what's going on and I think it's right now.

So are 3 new dec variables created, 2 for the metadata and the 3rd for the call to the decorator.

Reply:
Awesome! Seems good now, lgtm